front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/tag/TaggedEntity.groovy

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.front50.model.tag
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.netflix.spinnaker.front50.model.Timestamped
+
+class TaggedEntity implements Timestamped {
+  // {entityType}/{entityId}/{account}/{region}
+  String id
+
+  Long lastModified
+  String lastModifiedBy
+
+  Map<String, String> tags = [:]
+  EntityRef entityRef
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  static class EntityRef {
+    String entityType
+    String entityId
+
+    String cloudProvider
+    String application
+    String account
+    String region
```

Comment:
An `EntityRef` is an encapsulation of the coordinates needed to lookup tagged object.

A tagged object could be anything:
- cluster
- server group
- iam role
- load balancer
- security group

Each of these would have slightly different coordinates and at a minimum some combination of:
- account
- region
- cloudprovider

I'm not sure the granularity of things you would want to tag in GCP ... @duftler? 

I'm considering changing `EntityRef` to use `@JsonAnyGetter`/`@JsonAnySetter` and removing the explicit attributes.